### PR TITLE
Implemented Markdown

### PR DESCRIPTION
**Pull request by echen94**

### CODE REVIEW

File: file_handler.cc [hunk 1]
```diff
@@ -77,9 +78,40 @@ RequestHandler::Status StaticHandler::HandleRequest(const Request& request,
           content.append(buf, is.gcount());
 
       response->SetStatus(Response::ok);
-      response->AddHeader("content-type", mime.extension_to_type(extension));
-      response->AddHeader("content-length", std::to_string(content.length()));
-      response->SetBody(content);
+			//get the content type
```
Comment (neosbob) [inline file_handler.cc]: Lines 81 to 114, inconsistent spaces and tabs.

File: mime_types.cc [hunk 2]
```diff
@@ -12,7 +12,8 @@ struct mapping
   { "html", "text/html" },
   { "jpg", "image/jpeg" },
   { "png", "image/png" },
-  { "jpeg", "image/jpeg"}
+  { "jpeg", "image/jpeg"},
+	{ "md", "text/markdown"}
```
Comment (neosbob) [inline mime_types.cc]: Should fix alignment on spaces for line 16.

File: file_handler.cc [hunk 3]
```diff
@@ -77,9 +78,40 @@ RequestHandler::Status StaticHandler::HandleRequest(const Request& request,
...
+			//set response body
+			if (extension_type == "text/markdown")
+			{
+				markdown::Document doc;
+				doc.read(content);
+				std::ostringstream stream;
+				doc.write(stream);
+
+				std::string markdown = stream.str();
+
```
Comment (yanzunh) [inline file_handler.cc]: Some empty lines here is not necessary. Consider removing some of them

File: Makefile [hunk 4]
```diff
@@ -47,7 +47,7 @@ mime_types.o: mime_types.cc mime_types.h
 	g++ -c -std=c++11 mime_types.cc
 
 webserver: webserver.h webserver.cc webserver_main.cc config_parser.h config_parser.cc session.h request_handler.o session.o mime_types.o file_handler.o echo_handler.o not_found_handler.o status_handler.o request.o response.o http_client.o proxy_handler.o
-	g++ webserver.h webserver.cc webserver_main.cc config_parser.cc request_handler.o session.o mime_types.o file_handler.o echo_handler.o not_found_handler.o status_handler.o request.o response.o http_client.o proxy_handler.o -I /usr/local/Cellar/boost/1.54.0/include -std=c++11 -static-libgcc -static-libstdc++ -pthread -Wl,-Bstatic -lboost_thread -lboost_system -o webserver
+	g++ webserver.h webserver.cc webserver_main.cc config_parser.cc request_handler.o session.o mime_types.o file_handler.o echo_handler.o not_found_handler.o status_handler.o request.o response.o http_client.o proxy_handler.o -I cpp-markdown/*.cpp -I /usr/local/Cellar/boost/1.54.0/include -std=c++11 -static-libgcc -static-libstdc++ -pthread -Wl,-Bstatic -lboost_thread -lboost_system -lboost_regex -o webserver
```
Comment (yanzunh) [inline Makefile:50]: Not sure if it's necessary, but there are two "-I" flags here

**Review by yanzunh — APPROVED** [review]
Nicely done! Just some simple comment about styles

File: file_handler.cc (hunk 1 above)
Comment (echen94) [reply file_handler.cc]: Thanks for pointing that out! Will fix.

File: mime_types.cc (hunk 2 above)
Comment (echen94) [reply mime_types.cc]: Got it.

File: file_handler.cc (hunk 3 above)
Comment (echen94) [reply file_handler.cc]: Will delete the extra lines in my next commit.

**Pull request merged**